contracts/test/integration/optimisticPods/OptimisticPodTest.t.sol

```diff
@@ -0,0 +1,139 @@
+import "hardhat/console.sol";
+
+/// @dev Tests for the optimistic governance pod unit. This is composed of an
+/// Orca pod and an optimistic timelock. Key agents involved are:
+/// - Orca pod: a Gnosis Safe with a membership wrapper.
+///             It is the Gnosis safe from which transactions are sent to
+///             the optimistic timelock
+/// - Optimistic timelock: a timelock from which transactions are sent to the protocol
+contract OptimisticPodIntegrationTest is DSTest {
+    Vm public constant vm = Vm(HEVM_ADDRESS);
+
+    // Mainnet addresses
+    address private core = 0x8d5ED43dCa8C2F7dFB20CF7b53CC7E593635d7b9;
+    address private podControllerAddress =
+        0xD89AAd5348A34E440E72f5F596De4fA7e291A3e8;
+    address private memberTokenAddress =
+        0x0762aA185b6ed2dCA77945Ebe92De705e0C37AE3;
+
+    IMemberToken memberToken = IMemberToken(memberTokenAddress);
+
+    IControllerV1 controller = IControllerV1(podControllerAddress);
+
+    address proposer = address(0x1);
+    address executor = address(0x2);
+    address podAdmin = address(0x3);
+
+    address member1 = address(0x4);
+    address member2 = address(0x5);
+    address member3 = address(0x6);
+
+    uint256 podId;
+    uint256 numPodMembers;
+
+    function setUp() public {
+        mintOrcaTokens(address(this), 1, vm);
+
+        // Note: Gnosis safe creation fails if < 3 members
+        address[] memory members = new address[](3);
+        members[0] = member1;
+        members[1] = member2;
+        members[2] = member3;
+
+        numPodMembers = members.length;
+
+        podId = createPod(controller, memberToken, members, podAdmin);
+    }
+
+    /// @notice Validate that a pod can be created
+    function testCreatePod() public {
+        address safeAddress = controller.podIdToSafe(podId);
+        assert(safeAddress != address(0));
+
+        // Validate membership
+        address[] memory members = IGnosisSafe(safeAddress).getOwners();
+        assertEq(members.length, 3);
+
+        address setPodAdmin = controller.podAdmin(podId);
+        assertEq(setPodAdmin, podAdmin);
+    }
+
+    /// @notice Verify that the pod is set as the proposer and can propose on the timelock
+    function testLinkOptimisticTimelock() public {
+        address safeAddress = controller.podIdToSafe(podId);
+        OptimisticTimelock timelock = setupOptimisticTimelock(
+            safeAddress,
+            safeAddress,
+            core
+        );
+
+        // Be able to call propose via pod/safe. This verifies that the safe has onlyPropose role
+        vm.prank(safeAddress);
+        timelock.schedule(
+            address(0x10),
+            5,
+            "",
+            bytes32("testing"),
+            bytes32("random"),
+            0
+        );
+    }
+
+    /// @notice Validate that a member can be removed from a pod by the admin
+    /// Note: Orca sets the memberTokenID to be the podId
+    function testRemovePodMember() public {
```

Comment:
how would the DAO remove a pod member?

Reply:
PodAdmins can remove pod members. 

The DAO can remove members from the `TribalCouncil` by calling `memberToken.burn(memerToRemove, podId)`. 

The `TribalCouncil` in return is the `podAdmin` of child pods. It in turn can remove any member from a childPod by calling the burn function. 

There isn't a direct way at the moment for the DAO to remove members from child pods - it's devolved in this implementation to the `TribalCouncil`. Is that something we want to add?

Reply:
The DAO absolutely should have the ability to remove members of child pods imo

Reply:
This is implemented here: https://github.com/fei-protocol/fei-protocol-core/pull/587